x-pack/filebeat/docs/inputs/input-httpjson.asciidoc

```diff
@@ -1091,7 +1091,162 @@ request_url using file_name as 'file_2': https://example.com/services/data/v1.0/
 +
 Collect and make events from response in any format supported by httpjson for all calls.
 
-NOTE: httpjson chain will only create and ingest events from last call on chained configurations. Also, the current chain only supports the following: `request.url`, `request.method`, `request.transforms`, `request.body`, `response.transforms` and `response.split`.
+[float]
+==== `chain[].while`
+
+Contains basic request and response configuration for chained while calls. Chained while calls will keep making the requests for a given number of times until a condition is met
+or the maximum number of attempts gets exhausted. While chain has an attribute `until` which holds the expression to be evaluated. Ideally the `until` field should always be used 
+together with the attributes `request.retry.max_attempts` and `request.retry.wait_min` which specifies the maximum number of attempts to evaluate `until` before giving up and the 
+maximum wait time in between such requests. If `request.retry.max_attempts` is not specified, it will only try to evaluate the expression once and give up if it fails. If 
+`request.retry.wait_min` is not specified the default wait time will always be `0` as in successive calls will be made immediately.
+
+[float]
+==== `chain[].while.request`
+
+Please refer to <<request-parameters,request parameters>> for more information. Required.
+
+Example:
+
+First call: http://example.com/services/data/v1.0/exports
+
+Second call: http://example.com/services/data/v1.0/9ef0e6a5/export_ids/status
+
+Third call: http://example.com/services/data/v1.0/export_ids/1/info
+
+[float]
+==== `chain[].while.response.split`
+
+Please refer <<response-split,response split parameter>>.
+
+[float]
+==== `chain[].while.replace`
+
+Please refer `chain[].step.replace`.
```

Comment:
This PR still adds a few _"Please refer X"_ expressions. My understanding is that those are not correct, please correct me if I'm wrong.

I lean towards using _"See X"_ instead of _"Please refer to X for more information"_ as a replacement.

Also, this one uses a \`literal\` instead of a \<\<link\>\>, can it be turned into a link?

Reply:
@adriansr updated the docs as per the suggestions. Please review once.